c2corg_api/views/document_changes.py

```diff
@@ -0,0 +1,160 @@
+import datetime
+from operator import or_, and_
+
+from c2corg_api.models import DBSession
+from c2corg_api.models.document import Document, ArchiveDocumentLocale
+from c2corg_api.models.document_history import DocumentVersion, HistoryMetaData
+from c2corg_api.models.user import User
+from c2corg_api.views import cors_policy
+from c2corg_api.views.feed import get_params
+from c2corg_api.views.validation import validate_token_pagination,\
+  validate_user_id_not_required
+
+from sqlalchemy.sql.expression import desc
+
+from cornice.resource import resource, view
+from sqlalchemy.orm import joinedload, load_only
+from urllib import parse as urllib_parse
+
+
+@resource(path='/document/changes', cors_policy=cors_policy)
+class ChangesDocumentRest(object):
+    """ Unique class for returning history of a changes in documents.
+    """
+
+    def __init__(self, request):
+        self.request = request
+
+    # validate_pagination, validate_token_pagination, validate_user_id
+
+    @view(validators=[validate_token_pagination,
+                      validate_user_id_not_required])
+    def get(self):
+        """Get the public document changes feed (available to moderators).
+
+        Request:
+            `GET` `/document/changes[&limit=...][&token=...][&u=]`
+
+        Parameters:
+
+            `limit=...` (optional)
+            How many entries should be returned (default: 10).
+            The maximum is 50.
+
+            `token=...` (optional)
+            The pagination token. When requesting a feed, the response includes
+            a `pagination_token`. This token is to be used to request the next
+            page.
+
+            `u=...` (optional)
+            Changes made by one user.
+
+            For more information about "continuation token pagination", see:
+            http://www.servicedenuages.fr/pagination-continuation-token (fr)
+        """
+        user_id = self.request.validated.get('u')
+        lang, token_id, token_time, limit = get_params(self.request)
+
+        if token_time is None:
+            token_time = datetime.datetime.now()
+
+        changes = get_changes_of_feed(token_id, token_time, limit, user_id)
+
+        doc_ids = []
+        for ch in changes:
+            doc_ids.append(ch.id)
+
+        return load_feed(doc_ids, limit, user_id)
+
+
+def get_changes_of_feed(token_id, token_time, limit, user_id=None):
+    query = DBSession. \
+        query(HistoryMetaData.id, HistoryMetaData.user_id,
+              HistoryMetaData.written_at). \
+        order_by(HistoryMetaData.written_at.desc())
+
+    # pagination filter
+    if token_id is not None and token_time:
```

Comment:
It should be enough to use `HistoryMetaData.id` as pagination token. No need to also use `HistoryMetaData.written_at` because the time is set automatically when inserting new rows.